spec/lib/jenkins_pipeline_builder/generator_spec.rb

```diff
@@ -89,31 +89,52 @@ def fixture_path(fixture)
+        parsed_url = URI.parse('https://www.test.com')
+        file_contents = Zlib::GzipReader.new(File.open(tar_path)).read
+        file_object = double
+        allow(URI).to receive(:parse).and_return(parsed_url)
+        allow(parsed_url).to receive(:open).and_yield('A String')
+        allow(Zlib::GzipReader).to receive(:new).and_return(file_object)
+        allow(file_object).to receive(:read).and_return(file_contents)
+      end
 
-    it 'produces no errors while creating pipeline TemplatePipeline_nested' do
-      errors = bootstrap(fixture_path('template_pipeline_nested'), 'TemplatePipeline_nested')
-      expect(errors).to be_empty
-    end
+      it 'produces no errors while creating pipeline TemplatePipeline' do
+        errors = bootstrap(fixture_path('template_pipeline'), 'TemplatePipeline')
+        expect(errors).to be_empty
+      end
 
-    it 'loads extensions in remote dependencies' do
-      errors = bootstrap(fixture_path('template_pipeline'), 'TemplatePipeline')
-      expect(errors).to be_empty
-      expect(@generator.module_registry.registry[:job][:wrappers].keys).to include :test_wrapper
-      @generator.module_registry.registry[:job][:wrappers].delete(:test_wrapper)
-    end
+      it 'overrides the remote dependencies with local ones' do
+        errors = bootstrap(fixture_path('local_override/remote_and_local'), 'TemplatePipeline')
+        expect(errors).to be_empty
+        expect(@generator.job_collection.collection['{{name}}-10'][:value][:description]).to eq('Overridden stuff')
+      end
 
-    it 'overrides the remote dependencies with local ones' do
-      errors = bootstrap(fixture_path('local_override/remote_and_local'), 'TemplatePipeline')
-      expect(errors).to be_empty
-      expect(@generator.job_collection.collection['{{name}}-10'][:value][:description]).to eq('Overridden stuff')
-    end
+      it 'loads extensions in remote dependencies' do
+        errors = bootstrap(fixture_path('template_pipeline'), 'TemplatePipeline')
+        expect(errors).to be_empty
+        expect(@generator.module_registry.registry[:job][:wrappers].keys).to include :test_wrapper
+        @generator.module_registry.registry[:job][:wrappers].delete(:test_wrapper)
+      end
+
+      it 'fails to override when there are duplicate local items' do
+        expect { bootstrap(fixture_path('local_override/all_local'), 'TemplatePipeline') }.to raise_error(StandardError)
+      end
 
-    it 'fails to override when there are duplicate local items' do
-      expect { bootstrap(fixture_path('local_override/all_local'), 'TemplatePipeline') }.to raise_error(StandardError)
+      it 'produces no errors while creating pipeline TemplatePipeline_nested' do
+        tar_path = File.join(__dir__, 'fixtures/generator_tests/template_pipeline_nested/jobs.tar.gz')
+        parsed_url = URI.parse('https://www.test.com')
+        file_contents = Zlib::GzipReader.new(File.open(tar_path)).read
```

Comment:
It looks like you've factored the Gzip mocking out into the before clause. Can you remove it here?

Reply:
I believe we need to use it here since the tar path is changed slightly for this test case, and that means we have to restate the other mocks that rely on tar_path's value. But I have removed the variables and mocks that are not affected by tar_path changing!